core/src/main/java/com/linecorp/armeria/client/proxy/ProxyConfig.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.proxy;
+
+import static com.linecorp.armeria.client.proxy.DisabledProxyConfig.DISABLED_PROXY_CONFIG;
+import static java.util.Objects.requireNonNull;
+
+import java.net.InetSocketAddress;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.ClientFactory;
+
+/**
+ * Base configuration for proxy settings used by {@link ClientFactory}.
+ */
+public abstract class ProxyConfig {
+
+    /**
+     * Creates a {@code ProxyConfig} configuration for SOCKS4 protocol.
+     *
+     * @param proxyAddress The proxy address.
```

Comment:
nit but global comment:
```suggestion
     * @param proxyAddress The proxy address.
```

Reply:
I finally catched the difference! 🤣
![image](https://user-images.githubusercontent.com/1866157/76516063-21a08600-649e-11ea-89c5-825a92e60fd9.png)

Reply:
Oops, What I meant is, `The` should start with a lowercase. :sweat_smile:

Reply:
> I finally caught the difference! 

🦅👁.    😆😆😆